Test failures in Operator CI

### Describe the bug

Sometimes, the Keycloak Operator CI GitHub action fails due to failure in test `KeycloakDeploymentE2EIT.testBasicKeycloakDeploymentAndDeletion`. The GH action is considered as required for each PR and it's not possible to merge those PRs when the failure occurs. 

I noticed that in these PRs:
#10380 (specific [run](https://github.com/keycloak/keycloak/runs/5287572927?check_suite_focus=true))
#10396 (specific [run](https://github.com/keycloak/keycloak/runs/5290820646?check_suite_focus=true))
#10215 (specific [run](https://github.com/keycloak/keycloak/runs/5299598568?check_suite_focus=true))
...

@vmuzikar Could you please check it? Thanks.

```
INFO: Stopping Operator
2022-02-23 07:04:27,293 INFO  [io.jav.ope.Operator] (main) Operator SDK 2.0.2 is shutting down...
2022-02-23 07:04:27,300 INFO  [io.jav.ope.pro.eve.sou.con.ControllerResourceEventSource] (main) Stopping informer 'keycloakrealmimportcontroller' Controller -> io.fabric8.kubernetes.client.informers.impl.DefaultSharedIndexInformer@81a193b
2022-02-23 07:04:27,300 INFO  [io.jav.ope.pro.eve.sou.con.ControllerResourceEventSource] (ForkJoinPool.commonPool-worker-1) Stopping informer 'keycloakcontroller' Controller -> io.fabric8.kubernetes.client.informers.impl.DefaultSharedIndexInformer@20eb7cd0
Feb 23, 2022 7:04:27 AM org.keycloak.operator.OperatorManagedResource lambda$createOrUpdateReconciled$0
ERROR: Failed to create or update resource
Feb 23, 2022 7:04:27 AM org.keycloak.operator.OperatorManagedResource lambda$createOrUpdateReconciled$0
ERROR: ---
apiVersion: "apps/v1"
kind: "Deployment"
2022-02-23 07:04:27,410 ERROR [io.jav.ope.pro.eve.ReconciliationDispatcher] (EventHandler-keycloakcontroller) Error during error status handling.: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PUT at: https://192.168.49.2:8443/apis/keycloak.org/v2alpha1/namespaces/keycloak-test-c5af76ee-f7b3-4cd0-8f5e-e77bbedba5a1/keycloaks/example-kc/status. Message: Operation cannot be fulfilled on keycloaks.keycloak.org "example-kc": StorageError: invalid object, Code: 4, Key: /registry/keycloak.org/keycloaks/keycloak-test-c5af76ee-f7b3-4cd0-8f5e-e77bbedba5a1/example-kc, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: d7d8e402-f18e-4f7c-8c0e-8ad72a41952b, UID in object meta: . Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=keycloak.org, kind=keycloaks, name=example-kc, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Operation cannot be fulfilled on keycloaks.keycloak.org "example-kc": StorageError: invalid object, Code: 4, Key: /registry/keycloak.org/keycloaks/keycloak-test-c5af76ee-f7b3-4cd0-8f5e-e77bbedba5a1/example-kc, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: d7d8e402-f18e-4f7c-8c0e-8ad72a41952b, UID in object meta: , metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Conflict, status=Failure, additionalProperties={}).
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: "2022-02-23T07:04:08Z"
  generation: 3
  labels:
    address: "EvergreenTerrace742"
    app: "keycloak"
    app.kubernetes.io/managed-by: "keycloak-operator"
  managedFields:
  - apiVersion: "apps/v1"
    fieldsType: "FieldsV1"
    fieldsV1:
      f:metadata:
        f:annotations: {}
      f:status:
        f:availableReplicas: {}
        f:conditions:
          ".": {}
          k:{"type":"Available"}:
            ".": {}
            f:lastTransitionTime: {}
            f:lastUpdateTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
          k:{"type":"Progressing"}:
            ".": {}
            f:lastTransitionTime: {}
            f:lastUpdateTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
        f:observedGeneration: {}
        f:readyReplicas: {}
    manager: "kube-controller-manager"
    operation: "Update"
    subresource: "status"
    time: "2022-02-23T07:04:24Z"
  - apiVersion: "apps/v1"
    fieldsType: "FieldsV1"
    fieldsV1:
      f:metadata:
        f:annotations:
          f:deployment.kubernetes.io/revision: {}
        f:labels:
          ".": {}
          f:address: {}
          f:app: {}
          f:app.kubernetes.io/managed-by: {}
        f:ownerReferences:
          ".": {}
          k:{"uid":"d7d8e402-f18e-4f7c-8c0e-8ad72a41952b"}: {}
      f:spec:
        f:progressDeadlineSeconds: {}
        f:replicas: {}
        f:revisionHistoryLimit: {}
        f:selector: {}
        f:strategy:
          f:rollingUpdate:
            ".": {}
            f:maxSurge: {}
            f:maxUnavailable: {}
          f:type: {}
        f:template:
          f:metadata:
            f:labels:
              ".": {}
              f:app: {}
              f:app.kubernetes.io/managed-by: {}
          f:spec:
            f:containers:
              k:{"name":"keycloak"}:
                ".": {}
                f:args: {}
                f:env:
                  ".": {}
                  k:{"name":"KC_DB"}:
                    ".": {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"KC_DB_PASSWORD"}:
                    ".": {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"KC_DB_URL_HOST"}:
                    ".": {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"KC_DB_USERNAME"}:
                    ".": {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"KC_METRICS_ENABLED"}:
                    ".": {}
                    f:name: {}
                    f:value: {}
                f:image: {}
                f:imagePullPolicy: {}
                f:livenessProbe:
                  ".": {}
                  f:failureThreshold: {}
                  f:httpGet:
                    ".": {}
                    f:path: {}
                    f:port: {}
                    f:scheme: {}
                  f:initialDelaySeconds: {}
                  f:periodSeconds: {}
                  f:successThreshold: {}
                  f:timeoutSeconds: {}
                f:name: {}
                f:ports:
                  ".": {}
                  k:{"containerPort":8080,"protocol":"TCP"}:
                    ".": {}
                    f:containerPort: {}
                    f:protocol: {}
                  k:{"containerPort":8443,"protocol":"TCP"}:
                    ".": {}
                    f:containerPort: {}
                    f:protocol: {}
                f:readinessProbe:
                  ".": {}
                  f:failureThreshold: {}
                  f:httpGet:
                    ".": {}
                    f:path: {}
                    f:port: {}
                    f:scheme: {}
                  f:initialDelaySeconds: {}
                  f:periodSeconds: {}
                  f:successThreshold: {}
                  f:timeoutSeconds: {}
                f:resources: {}
                f:terminationMessagePath: {}
                f:terminationMessagePolicy: {}
            f:dnsPolicy: {}
            f:restartPolicy: {}
            f:schedulerName: {}
            f:securityContext: {}
            f:terminationGracePeriodSeconds: {}
    manager: "okhttp"
    operation: "Update"
    time: "2022-02-23T07:04:26Z"
  name: "example-kc"
  namespace: "keycloak-test-c5af76ee-f7b3-4cd0-8f5e-e77bbedba5a1"
  ownerReferences:
  - apiVersion: "keycloak.org/v2alpha1"
    kind: "Keycloak"
    blockOwnerDeletion: true
    controller: true
    name: "example-kc"
    uid: "d7d8e402-f18e-4f7c-8c0e-8ad72a41952b"
  resourceVersion: "799"
  uid: "ad326fab-46e4-4c2c-8b03-63ed57d72a24"
spec:
  replicas: 1
  selector:
    matchLabels:
      app: "keycloak"
      app.kubernetes.io/managed-by: "keycloak-operator"
  strategy:
    rollingUpdate:
      maxSurge: "25%"
      maxUnavailable: "25%"
    type: "RollingUpdate"
  template:
    metadata:
      labels:
        app: "keycloak"
        app.kubernetes.io/managed-by: "keycloak-operator"
    spec:
      containers:
      - args:
        - "start-dev"
        env:
        - name: "KC_DB_PASSWORD"
          value: "testpassword"
        - name: "KC_DB_USERNAME"
          value: "postgres"
        - name: "KC_DB"
          value: "postgres"
        - name: "KC_DB_URL_HOST"
          value: "postgres-db"
        - name: "KC_METRICS_ENABLED"
          value: "true"
        image: "keycloak:8610d61c835dcaf165d336c17038c3e3494aa9be"
        imagePullPolicy: "Never"
        livenessProbe:
          failureThreshold: 100
          httpGet:
            path: "/health/live"
            port: 8080
          initialDelaySeconds: 15
          periodSeconds: 2
        name: "keycloak"
        ports:
        - containerPort: 8443
          protocol: "TCP"
        - containerPort: 8080
          protocol: "TCP"
        readinessProbe:
          failureThreshold: 200
          httpGet:
            path: "/health/ready"
            port: 8080
          initialDelaySeconds: 15
          periodSeconds: 2
      dnsPolicy: "ClusterFirst"
      restartPolicy: "Always"
      terminationGracePeriodSeconds: 30
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2022-02-23T07:04:24Z"
    lastUpdateTime: "2022-02-23T07:04:24Z"
    message: "Deployment has minimum availability."
    reason: "MinimumReplicasAvailable"
    status: "True"
    type: "Available"
  - lastTransitionTime: "2022-02-23T07:04:08Z"
    lastUpdateTime: "2022-02-23T07:04:25Z"
    message: "ReplicaSet \"example-kc-c7cb69c55\" is progressing."
    reason: "ReplicaSetUpdated"
    status: "True"
    type: "Progressing"
  observedGeneration: 2
  readyReplicas: 1
  replicas: 2
  unavailableReplicas: 1
  updatedReplicas: 1

Feb 23, 2022 7:04:27 AM org.keycloak.operator.v2alpha1.KeycloakController updateErrorStatus
ERROR: --- Error reconciling
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://192.168.49.2:8443/apis/apps/v1/namespaces/keycloak-test-c5af76ee-f7b3-4cd0-8f5e-e77bbedba5a1/deployments/example-kc. Message: Not Found.
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:683)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:662)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:613)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:556)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:519)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleGet(OperationSupport.java:488)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleGet(OperationSupport.java:458)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleGet(BaseOperation.java:696)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.getMandatory(BaseOperation.java:182)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.require(BaseOperation.java:163)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation.requireFromServer(HasMetadataOperation.java:120)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation.replace(HasMetadataOperation.java:187)
	at io.fabric8.kubernetes.client.dsl.base.HasMetadataOperation.replace(HasMetadataOperation.java:141)
	at io.fabric8.kubernetes.client.dsl.internal.apps.v1.RollableScalableResourceOperation.replace(RollableScalableResourceOperation.java:152)
	at io.fabric8.kubernetes.client.dsl.internal.apps.v1.DeploymentOperationsImpl.replace(DeploymentOperationsImpl.java:116)
	at io.fabric8.kubernetes.client.dsl.internal.apps.v1.DeploymentOperationsImpl.replace(DeploymentOperationsImpl.java:54)
	at io.fabric8.kubernetes.client.utils.CreateOrReplaceHelper.replace(CreateOrReplaceHelper.java:68)
	at io.fabric8.kubernetes.client.utils.CreateOrReplaceHelper.createOrReplace(CreateOrReplaceHelper.java:60)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.createOrReplace(BaseOperation.java:316)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.createOrReplace(BaseOperation.java:83)
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.createOrReplace(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java:105)
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.createOrReplace(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java:63)
	at org.keycloak.operator.OperatorManagedResource.lambda$createOrUpdateReconciled$0(OperatorManagedResource.java:56)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at org.keycloak.operator.OperatorManagedResource.createOrUpdateReconciled(OperatorManagedResource.java:50)
	at org.keycloak.operator.v2alpha1.KeycloakController.reconcile(KeycloakController.java:83)
	at org.keycloak.operator.v2alpha1.KeycloakController.reconcile(KeycloakController.java:49)
	at org.keycloak.operator.v2alpha1.KeycloakController_ClientProxy.reconcile(Unknown Source)
	at io.javaoperatorsdk.operator.processing.Controller$2.execute(Controller.java:101)
	at io.javaoperatorsdk.operator.processing.Controller$2.execute(Controller.java:76)
	at io.javaoperatorsdk.operator.api.monitoring.Metrics.timeControllerExecution(Metrics.java:34)
	at io.javaoperatorsdk.operator.processing.Controller.reconcile(Controller.java:75)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.reconcileExecution(ReconciliationDispatcher.java:151)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleReconcile(ReconciliationDispatcher.java:117)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleDispatch(ReconciliationDispatcher.java:82)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleExecution(ReconciliationDispatcher.java:51)
	at io.javaoperatorsdk.operator.processing.event.EventProcessor$ControllerExecution.run(EventProcessor.java:385)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:[829](https://github.com/keycloak/keycloak/runs/5299598568?check_suite_focus=true#step:9:829))

	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:683)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:662)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:613)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:556)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:519)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleUpdate(OperationSupport.java:343)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleUpdate(OperationSupport.java:323)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleUpdate(BaseOperation.java:647)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.updateStatus(BaseOperation.java:502)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.updateStatus(BaseOperation.java:83)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher$CustomResourceFacade.updateStatus(ReconciliationDispatcher.java:328)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleErrorStatusHandler(ReconciliationDispatcher.java:191)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleReconcile(ReconciliationDispatcher.java:119)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleDispatch(ReconciliationDispatcher.java:82)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleExecution(ReconciliationDispatcher.java:51)
	at io.javaoperatorsdk.operator.processing.event.EventProcessor$ControllerExecution.run(EventProcessor.java:385)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

2022-02-23 07:04:27,410 INFO  [io.jav.ope.pro.eve.ReconciliationDispatcher] (EventHandler-keycloakcontroller) Kubernetes exception 404 Failure executing: GET at: https://192.168.49.2:[844](https://github.com/keycloak/keycloak/runs/5299598568?check_suite_focus=true#step:9:844)3/apis/apps/v1/namespaces/keycloak-test-c5af76ee-f7b3-4cd0-8f5e-e77bbedba5a1/deployments/example-kc. Message: Not Found. during event processing, ExecutionScope{ resource id: CustomResourceID{name='example-kc', namespace='keycloak-test-c5af76ee-f7b3-4cd0-8f5e-e77bbedba5a1'}, version: 780} failed
Feb 23, 2022 7:04:27 AM org.keycloak.operator.ClusterOperatorTest after
INFO: Creating new K8s Client
Feb 23, 2022 7:04:27 AM org.keycloak.operator.ClusterOperatorTest after
INFO: Deleting namespace : keycloak-test-c5af76ee-f7b3-4cd0-8f5e-e77bbedba5a1
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 115.331 s <<< FAILURE! - in org.keycloak.operator.KeycloakDeploymentE2EIT
Error:  testBasicKeycloakDeploymentAndDeletion  Time elapsed: 1.132 s  <<< FAILURE!
java.lang.AssertionError: 

Expecting actual not to be null
	at org.keycloak.operator.KeycloakDeploymentE2EIT.testBasicKeycloakDeploymentAndDeletion(KeycloakDeploymentE2EIT.java:38)
```

Final results:
```
Error:  Failures: 
Error:    KeycloakDeploymentE2EIT.testBasicKeycloakDeploymentAndDeletion:38 
Expecting actual not to be null
[INFO] 
Error:  Tests run: 5, Failures: 1, Errors: 0, Skipped: 0
```

### Expected behavior

Successful execution of those tests.

### Actual behavior

Present failure for test `KeycloakDeploymentE2EIT`.

--- a/operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java
+++ b/operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java
@@ -15,8 +15,10 @@ import io.quarkus.logging.Log;
 import org.awaitility.Awaitility;
 import org.eclipse.microprofile.config.ConfigProvider;
 import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
 
 import javax.enterprise.inject.Instance;
 import javax.enterprise.inject.spi.CDI;
@@ -28,6 +30,7 @@ import java.io.FileWriter;
 import java.time.Duration;
 import java.util.List;
 import java.util.UUID;
+import java.util.concurrent.TimeUnit;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
@@ -61,10 +64,10 @@ public abstract class ClusterOperatorTest {
     setDefaultAwaitilityTimings();
     calculateNamespace();
     createK8sClient();
+    createCRDs();
     createNamespace();
 
     if (operatorDeployment == OperatorDeployment.remote) {
-      createCRDs();
       createRBACresourcesAndOperatorDeployment();
     } else {
       createOperator();
@@ -107,8 +110,17 @@ public abstract class ClusterOperatorTest {
   }
   private static void createCRDs() throws FileNotFoundException {
     Log.info("Creating CRDs");
-    k8sclient.load(new FileInputStream(TARGET_KUBERNETES_GENERATED_YML_FOLDER + "keycloaks.keycloak.org-v1.yml")).createOrReplace();
-    k8sclient.load(new FileInputStream(TARGET_KUBERNETES_GENERATED_YML_FOLDER + "keycloakrealmimports.keycloak.org-v1.yml")).createOrReplace();
+    try {
+      var deploymentCRD = k8sclient.load(new FileInputStream(TARGET_KUBERNETES_GENERATED_YML_FOLDER + "keycloaks.keycloak.org-v1.yml"));
+      deploymentCRD.createOrReplace();
+      deploymentCRD.waitUntilReady(5, TimeUnit.SECONDS);
+      var realmImportCRD = k8sclient.load(new FileInputStream(TARGET_KUBERNETES_GENERATED_YML_FOLDER + "keycloakrealmimports.keycloak.org-v1.yml"));
+      realmImportCRD.createOrReplace();
+      realmImportCRD.waitUntilReady(5, TimeUnit.SECONDS);
+    } catch (Exception e) {
+      Log.warn("Failed to create Keycloak CRD, retrying", e);
+      createCRDs();
+    }
   }
 
   private static void registerReconcilers() {
@@ -170,6 +182,23 @@ public abstract class ClusterOperatorTest {
   private static void setDefaultAwaitilityTimings() {
     Awaitility.setDefaultPollInterval(Duration.ofSeconds(1));
     Awaitility.setDefaultTimeout(Duration.ofSeconds(180));
+  }
+
+  @AfterEach
+  public void cleanup() {
+    Log.info("Deleting Keycloak CR");
+    k8sclient.resources(Keycloak.class).delete();
+    Awaitility.await()
+            .untilAsserted(() -> {
+              var kcDeployments = k8sclient
+                      .apps()
+                      .deployments()
+                      .inNamespace(namespace)
+                      .withLabels(Constants.DEFAULT_LABELS)
+                      .list()
+                      .getItems();
+              assertThat(kcDeployments.size()).isEqualTo(0);
+            });
   }
 
   @AfterAll

--- a/operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java
+++ b/operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java
@@ -5,7 +5,6 @@ import io.fabric8.kubernetes.api.model.apps.DeploymentSpecBuilder;
 import io.quarkus.logging.Log;
 import io.quarkus.test.junit.QuarkusTest;
 import org.awaitility.Awaitility;
-import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
 import org.keycloak.operator.v2alpha1.crds.Keycloak;
 
@@ -111,12 +110,6 @@ public class KeycloakDeploymentE2EIT extends ClusterOperatorTest {
             savePodLogs();
             throw e;
         }
-    }
-
-    @AfterEach
-    public void cleanup() {
-        Log.info("Deleting Keycloak CR");
-        k8sclient.resources(Keycloak.class).delete(getDefaultKeycloakDeployment());
     }
 
 }

